ee/clickhouse/sql/person.py

```diff
@@ -95,29 +102,40 @@
 PERSONS_DISTINCT_ID_TABLE_BASE_SQL = """
 CREATE TABLE {table_name} ON CLUSTER {cluster}
 (
-    id Int64,
     distinct_id VARCHAR,
     person_id UUID,
     team_id Int64,
-    is_deleted Int8 DEFAULT 0
+    _sign Int8 DEFAULT 1,
+    is_deleted Int8 ALIAS if(_sign==-1, 1, 0)
     {extra_fields}
 ) ENGINE = {engine}
 """
 
 PERSONS_DISTINCT_ID_TABLE_SQL = (
     PERSONS_DISTINCT_ID_TABLE_BASE_SQL
-    + """Order By (team_id, distinct_id, person_id, id)
+    + """Order By (team_id, distinct_id, person_id)
 {storage_policy}
 """
 ).format(
     table_name=PERSONS_DISTINCT_ID_TABLE,
     cluster=CLICKHOUSE_CLUSTER,
-    engine=table_engine(PERSONS_DISTINCT_ID_TABLE, "_timestamp", REPLACING_MERGE_TREE),
+    engine=table_engine(PERSONS_DISTINCT_ID_TABLE, "_sign", COLLAPSING_MERGE_TREE),
     extra_fields=KAFKA_COLUMNS,
     storage_policy=STORAGE_POLICY,
 )
 
-KAFKA_PERSONS_DISTINCT_ID_TABLE_SQL = PERSONS_DISTINCT_ID_TABLE_BASE_SQL.format(
+# :KLUDGE: We default is_deleted to 0 for backwards compatibility for when we drop `is_deleted` from message schema.
+#    Can't make DEFAULT if(_sign==-1, 1, 0) because Cyclic aliases error.
+KAFKA_PERSONS_DISTINCT_ID_TABLE_SQL = """
+CREATE TABLE {table_name} ON CLUSTER {cluster}
+(
+    distinct_id VARCHAR,
+    person_id UUID,
+    team_id Int64,
+    _sign Nullable(Int8),
+    is_deleted Nullable(Int8)
+) ENGINE = {engine}
+""".format(
     table_name="kafka_" + PERSONS_DISTINCT_ID_TABLE,
     cluster=CLICKHOUSE_CLUSTER,
     engine=kafka_engine(KAFKA_PERSON_UNIQUE_ID),
```

Comment:
@fuziontech this is the first time we're replacing the table on cloud outright. Will the default zookeeper paths give us trouble, thoughts on what to do there?

Reply:
It definitely doesn't matter for kafka table engines since the consumer group persists. 

We actually have dropped tables on cloud outright manually quite a few times. We just have to be careful with confirming that the ZK paths are removed before we create the new table which _usually_ happens immediately. That is unless there are thousands of mutations backed up.

Reply:
The comment went to the wrong place - I intended to ask about the `person_distinct_id` table rather :)

Re ` ZK paths are removed before we create the new table which usually happens immediately` - how to verify this? Want to make sure the default path set by out code does not clash for the distinct_id table